src/test/java/org/jenkinsci/plugins/docker/commons/credentials/DockerServerCredentialsBindingTest.java

```diff
@@ -0,0 +1,182 @@
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                DockerServerCredentials c = new DockerServerCredentials(CredentialsScope.GLOBAL,
+                        "docker-client-cert", "desc", "clientKey", "clientCertificate", "serverCaCertificate");
+                CredentialsProvider.lookupStores(story.j.jenkins).iterator().next().addCredentials(Domain.global(), c);
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                final String shellStep1, shellStep2;
+                if (Functions.isWindows()) {
+                    shellStep1 = ""
+                            + "    bat '''\n"
+                            + "      REM check existence of the credentials dir\n"
+                            + "      if not exist %DOCKER_CERT_PATH% exit /B 1\n"
+                            + "\n"
+                            + "      REM check existence of the certificate files\n"
+                            + "      if not exist %DOCKER_CERT_PATH%\\\\key.pem exit /B 1\n"
+                            + "      if not exist %DOCKER_CERT_PATH%\\\\cert.pem exit /B 1\n"
+                            + "      if not exist %DOCKER_CERT_PATH%\\\\ca.pem exit /B 1\n"
+                            + "\n"
+                            + "      REM keep location of the certificate dir for the next step\n"
+                            + "      echo %DOCKER_CERT_PATH% > cert-path\n"
+                            + "    '''\n";
+                    shellStep2 = ""
+                            + "  bat '''\n"
+                            + "    REM get path of the certificate directory\n"
+                            + "    if not exist cert-path exit /B 1\n"
+                            + "    set /p cert_path=<cert-path\n"
+                            + "\n"
+                            + "    REM check it has been deleted\n"
+                            + "    if exist %cert_path% exit /B 1\n"
+                            + "  '''\n";
+                } else {
+                    shellStep1 = ""
+                            + "    sh '''\n"
+                            + "      set -e -x\n"
+                            + "\n"
+                            + "      # check permissions on the credentials dir and its parent\n"
+                            + "      [ $(stat -c %a \"$DOCKER_CERT_PATH\")    = 700 ]\n"
+                            + "      [ $(stat -c %a \"$DOCKER_CERT_PATH\"/..) = 700 ]\n"
+                            + "\n"
+                            + "      # check permissions and content of the certificate files\n"
+                            + "      [ $(stat -c %a \"$DOCKER_CERT_PATH/key.pem\")  = 600 ]\n"
+                            + "      [ $(stat -c %a \"$DOCKER_CERT_PATH/cert.pem\") = 600 ]\n"
+                            + "      [ $(stat -c %a \"$DOCKER_CERT_PATH/ca.pem\")   = 600 ]\n"
```

Comment:
BTW this stuff could be put into `src/test/resources/` and copied prior to running the build using `TheClass.class.getResource[AsStream]` and `Node.getWorkspaceFor`. Might be more readable that way.